### PR TITLE
Fixed a bug with optional Config parameters

### PR DESCRIPTION
**Pull request by VPetukhov**
PR fixes https://github.com/comonicon/Comonicon.jl/issues/228 and adds tests that fail on the current version, but work fine with the fix.

### CODE REVIEW

File: test/codegen/julia/config.jl [hunk 1]
```diff
@@ -5,8 +5,8 @@ using Comonicon
 using Configurations
 
 @option struct OptionA
```
Comment (Roger-luo) [inline test/codegen/julia/config.jl:7]: can we have a separate test (in a separate test module) so we can have both tests (original and the new case)

**Review by Roger-luo — CHANGES_REQUESTED** [review]
Thanks! the PR looks good to me in general, but can we keep both tests? just don't want to remove any existing tests.

File: test/codegen/julia/config.jl (hunk 1 above)
Comment (VPetukhov) [reply test/codegen/julia/config.jl:7]: @Roger-luo , sure, I can move it to a separate files, but why? All the original tests are [still there](https://github.com/comonicon/Comonicon.jl/pull/229/files#diff-b0b97d6d43551d1c81128515a32a025f76026fa98bb961bca6fd7d2e739f1e80R33), I only replaced `@main` with `@cast` to allow multiple entry points for different tests.

Comment (Roger-luo) [reply test/codegen/julia/config.jl:7]: oh you are right, sry I misread the git diff of `OptionA`.

**Review by Roger-luo — APPROVED** [review]

File: src/codegen/julia.jl [hunk 2]
```diff
@@ -274,8 +274,10 @@ function emit_leaf_call(cmd::LeafCommand, ctx::EmitContext, args::Symbol, kwargs
         for option in values(cmd.options)
             Configurations.is_option(option.type) || continue
             push!(ret.args, quote
-                $kwargs[$(QuoteNode(option.sym))] =
-                    $Configurations.from_dict($(option.type), $kwargs[$(QuoteNode(option.sym))])
+                if $(QuoteNode(option.sym)) in Base.keys($kwargs)
```
Comment (Roger-luo) [inline src/codegen/julia.jl:277]:
```suggestion
                if $(QuoteNode(option.sym)) in $Base.keys($kwargs)
```

althought we have access to `Base` most of the time, but this won't work if it's a `baremodule` which does not have access to `Base`, so we need to insert the module value at compile time.

**Review by Roger-luo — APPROVED** [review]
I'm not able to edit this PR, but I'm gonna merge this PR first, and edit the module insertion myself.

**Pull request merged**